lib/rules/utils/ast-utils.js

```diff
@@ -853,14 +853,7 @@ module.exports = {
     isDirectiveComment(node) {
         const comment = node.value.trim();
 
-        return (
-            node.type === "Line" && comment.indexOf("eslint-") === 0 ||
-            node.type === "Block" && (
-                comment.indexOf("global ") === 0 ||
-                comment.indexOf("eslint ") === 0 ||
-                comment.indexOf("eslint-") === 0
-            )
-        );
```

Comment:
it's likely that some comments were wrongly treated as directives. e.g. `//eslint-xxxx`

Reply:
I think it's okay, this check wasn't designed to perfectly match directives. Worst case is that a comment that pretty much looks like a directive will be ignored by a rule (e.g. it will be allowed as an inline comment).

Reply:
And the exisiting tests like `// eslint-secret-directive` confirm that anything that starts with `eslint-` was supposed to match. :)

Reply:
But it was missing `globals` and `exported`:

```js
/* eslint no-inline-comments: error */

foo(); /* global ok */

foo(); /* globals not ok */

foo(); /* exported not ok */

```

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWlubGluZS1jb21tZW50czogZXJyb3IgKi9cblxuZm9vKCk7IC8qIGdsb2JhbCBvayAqL1xuXG5mb28oKTsgLyogZ2xvYmFscyBub3Qgb2sgKi9cblxuZm9vKCk7IC8qIGV4cG9ydGVkIG5vdCBvayAqL1xuXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

I think we account for them in the new regex.

Reply:
updated, will add some tests later.